zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/transfer/SnapshotTransfer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.snapshots.transfer;
+
+import io.camunda.zeebe.scheduler.ConcurrencyControl;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.snapshots.PersistedSnapshot;
+import io.camunda.zeebe.snapshots.ReceivableSnapshotStore;
+import io.camunda.zeebe.snapshots.ReceivedSnapshot;
+import io.camunda.zeebe.snapshots.SnapshotChunk;
+import io.camunda.zeebe.util.collection.Tuple;
+import java.util.UUID;
+
+/**
+ * Receive a complete snapshot from a {@link SnapshotTransferService} by repeatedly asking for the
+ * next chunk until all chunks are received. No retry is done on the futures, if you want support
+ * for retry, wrap {@param service} with retries.
+ *
+ * <p>Snapshots are received in the {@param snapshotStore}.
+ */
+public class SnapshotTransfer {
+
+  private final SnapshotTransferService service;
+  private final ReceivableSnapshotStore snapshotStore;
+  private final ConcurrencyControl concurrency;
+
+  public SnapshotTransfer(
+      final SnapshotTransferService service,
+      final ReceivableSnapshotStore snapshotStore,
+      final ConcurrencyControl concurrency) {
+    this.service = service;
+    this.snapshotStore = snapshotStore;
+    this.concurrency = concurrency;
+  }
+
+  public ActorFuture<PersistedSnapshot> getLatestSnapshot(final int partitionId) {
+    final var transferId = UUID.randomUUID();
+    return service
+        .getLatestSnapshot(partitionId, transferId)
+        .andThen(
+            snapshot -> {
+              if (snapshot == null) {
+                return CompletableActorFuture.completedExceptionally(
+                    new IllegalArgumentException(
```

Comment:
:question: Is illegal argument really the case here? I think a `SnapshotNotFoundException` might be more appropriate? Maybe?